packages/plugins/plugin-table/src/components/ObjectTable.tsx

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import React, { useCallback } from 'react';
+
+import { type EchoReactiveObject } from '@dxos/echo-schema';
+import { useGlobalFilteredObjects } from '@dxos/plugin-search';
+import { create, getSpace, useQuery, Filter } from '@dxos/react-client/echo';
+import { Button, Icon } from '@dxos/react-ui';
+
+import { Table } from './Table';
+import { useStarterTable } from '../hooks/useStarterTable';
+import { type TableType } from '../types';
+
+export type ObjectTableProps = {
+  table: TableType;
+};
+
+export const ObjectTable = ({ table }: ObjectTableProps) => {
+  const space = getSpace(table);
+  const queriedObjects = useQuery<EchoReactiveObject<any>>(
+    space,
+    table.schema ? Filter.schema(table.schema) : () => false,
+    undefined,
+    // TODO(burdon): Toggle deleted.
+    [table.schema],
+  );
+  const filteredObjects = useGlobalFilteredObjects(queriedObjects);
+
+  const handleNewRow = useCallback(() => {
+    if (!table.schema || !space) {
+      return;
+    }
+    space.db.add(create(table.schema, {}));
+  }, [table.schema, space]);
+
+  useStarterTable({ table, space });
```

Comment:
i think it's better to change this hook into a util (create default table) and then use an explicit useEffect here to set it (i.e., don't obfuscate that decision).

Reply:
Fair point.